### PR TITLE
Refactor pre code generation

### PR DESCRIPTION
**Pull request by german1608**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "TAC"]
+	path = TAC
```
Comment (aitorres) [inline .gitmodules]: Just a question: wouldn't it be better to have the TAC submodule under `src`?

Comment (german1608) [reply .gitmodules]: Yes, indeed

File: src/frontend/Errors.hs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module Errors where
+
+import Utils
+
+data Error = Error String Position
+    deriving Show
+
+data ErrorCategory = LexError | SemanticError
```
Comment (aitorres) [inline src/frontend/Errors.hs]: Shouldn't we also have ParserErrors?

Comment (german1608) [reply src/frontend/Errors.hs]: Yes, but that can be a new issue. Currently the parser log parser and semantic errors as if they are the same thing.

**Review by aitorres — APPROVED** [review]

File: src/frontend/Errors.hs (hunk 2 above)
Comment (aitorres) [reply src/frontend/Errors.hs]: Will do!

**Comment by aitorres** [comment]
This PR fixes #5

**Pull request merged**